articles/iot-operations/deploy-iot-ops/howto-manage-secrets.md

```diff
@@ -26,6 +26,111 @@ Azure IoT Operations supports Azure Key Vault for storing secrets and certificat
+
+1. On the resource page for your app, select **Certificates & secrets** from the **Manage** section of the app menu.
+
+1. Select **New client secret**.
+
+1. Provide an optional description for the secret, then select **Add**.
+
+1. Copy the **Value** and **Secret ID** from your new secret. You'll use these values later below.
+
+Retrieve the service principal Object Id
+
+1. On the **Overview** page for your app, under the section **Essentials**, click on the **Application name** link under **Managed application in local directory**. This opens the Enterprise Application properties. Copy the Object Id to use later.
+
+### Create an Azure Key Vault
+
+Create a new Azure Key Vault service and ensure it has the **Permission Model** to Vault access policy.
+
+```bash
+az keyvault create --enable-rbac-authorization false --name "<your unique key vault name>" --resource-group "<the name of the resource group>"
+```
+If you have an existing Key Vault, you can change the Permission Model by executing the following:
+
+```bash
+az keyvault update --name "<your unique key vault name>" --resource-group "<the name of the resource group>" --enable-rbac-authorization false 
+```
+You will need the Key Vault Resource ID below, to retrieve it run:
+
+```bash
+az keyvault show --name "<your unique key vault name>" --resource-group "<the name of the resource group>" --query id  -o tsv
+```
+
+### Set service principal access policy in Azue Key Vault
+
+The newly created service principal needs **Secret** `list` and `get` access policy for the Azure IoT Operations to work with the secret store. 
+
+Run the following to assign **secret** `get` and `list` permissions to the service principal.
+
+```bash
+
+az keyvault set-policy --name "<your unique key vault name>" --resource-group "<the name of the resource group>" --object-id <Object ID copied from Enterprise Application SP in Microsoft Entra ID> --secret-permissions get list --key-permissions get list
+
+```
+
+### Passing in service principal and Azure Key Vault arguments when running Azure IoT Operations deployment
+
+When following the guide [Deploy Azure IoT Operations extensions](./howto-deploy-iot-operations.md?tabs=cli), you will need to pass in additional flags to the `az iot ops init` command in order to use the pre-configured service principal and key vault.
+
+For example, the following shows how to prepare the cluster for Azure IoT Operations without fully deploying it by using `--no-deploy` flag. You can also run the command without this argument for a default Azure IoT Operations deployment.
+
+```bash
+
+az iot ops init --name "<your unique key vault name>" --resource-group "<the name of the resource group>" \
+    --kv-id <Key Vault Resource ID retrieved above> \
+    --sp-app-id <Application registration App ID (client ID) from Microsoft Entra ID above> \
+    --sp-object-id <Object ID copied from Enterprise Application in Microsoft Entra ID above> \
```

Comment:
```suggestion
    --kv-id <Key Vault Resource ID> \
    --sp-app-id <Application registration App ID (client ID) from Microsoft Entra ID> \
    --sp-object-id <Object ID copied from Enterprise Application in Microsoft Entra ID> \
```

info: in docs, we avoid directional terms like above and below. If you think it's necessary to refer to another section or step, then instead we can use temporal terms like previously, earlier, or later.